apps/prairielearn/src/components/QuestionNavigation.html.ts

```diff
@@ -0,0 +1,109 @@
+import { HtmlValue, html } from '@prairielearn/html';
+
+export function QuestionNavSideGroup({
+  urlPrefix,
+  prevInstanceQuestionId,
+  nextInstanceQuestionId,
+  sequenceLocked,
+  prevGroupRolePermissions,
+  nextGroupRolePermissions,
+  advanceScorePerc,
+  userGroupRoles,
+}: {
+  urlPrefix: string;
+  prevInstanceQuestionId: string;
+  nextInstanceQuestionId: string;
+  sequenceLocked: boolean | null;
+  prevGroupRolePermissions: { can_view?: boolean } | null;
+  nextGroupRolePermissions: { can_view?: boolean } | null;
+  advanceScorePerc: number | null;
+  userGroupRoles: string | null;
+}) {
+  return html`
+    <div class="text-center mb-2">
+      ${QuestionNavSideButton({
+        questionId: prevInstanceQuestionId,
+        urlPrefix,
+        button: { id: 'question-nav-prev', label: 'Previous question' },
+        groupRolePermissions: prevGroupRolePermissions,
+        userGroupRoles,
+      })}
+      ${QuestionNavSideButton({
+        // NOTE: This must be kept in sync the the corresponding code in
+        // `lib/question-render.js`.
+        questionId: nextInstanceQuestionId,
+        sequenceLocked,
+        urlPrefix,
+        button: { id: 'question-nav-next', label: 'Next question' },
+        groupRolePermissions: nextGroupRolePermissions,
+        advanceScorePerc,
+        userGroupRoles,
+      })}
+    </div>
+  `;
+}
+
+export function QuestionNavSideButton({
```

Comment:
The functionality of this function is expected to be similar to the original EJS, but the code was rewritten for clarity. A few changes were done to hopefully get the result better:

* The id and label are computed in the function itself instead of sent as parameters, to avoid the need for "synchronization" between different renderings of the button;
* If the button is disabled, it uses a button instead of a link (with early return);
* If the button shows a popover, it uses a button instead of a link (with early return);
* The group role test is checked before the sequence lock: if both restrictions are in place, the role will remain after an update in the score, so there's no point in showing the sequence warning.